pkg/plugins/gateway/base_http/http.go

```diff
@@ -40,8 +43,20 @@ type BaseHttpGateway struct {
 	mw HttpMiddleware
 }
 
-func (s *BaseHttpGateway) httpHandler(pool worker.WorkerPool) func(ctx *fasthttp.RequestCtx) {
+func (s *BaseHttpGateway) httpHandler(unWrappedPool worker.WorkerPool) func(ctx *fasthttp.RequestCtx) {
 	return func(ctx *fasthttp.RequestCtx) {
+		pool := &worker.InstrumentedWorkerPool{
+			WorkerPool: unWrappedPool,
+			Wrapper:    worker.InstrumentedWorkerFn(span.FromFastHttp(ctx), true, false),
+		}
```

Comment:
should we be re-wrapping this on every request? We can update the worker methods to accept context to potentially avoid this.